kornia/feature/disk/structs.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+
+import torch
+import torch.nn.functional as F
+
+from kornia.core import Tensor
+
+
+@dataclass
+class DISKFeatures:
+    r"""A data structure holding DISK keypoints, descriptors and detection scores for an image. Since DISK detects a
+    varying number of keypoints per image, `DISKFeatures` is not batched.
+
+    Args:
+        keypoints: Tensor of shape :math:`(N, 2)`, where :math:`N` is the number of keypoints.
+        descriptors: Tensor of shape :math:`(N, 128)`.
+        detection_scores: Tensor of shape :math:`(N,)` where the detection score can be interpreted as
+        the log-probability of keeping a keypoint after it has been proposed (see the paper,
+        section Method->Feature distribution for details).
+    """
+    keypoints: Tensor
+    descriptors: Tensor
+    detection_scores: Tensor
+
+    @property
+    def n(self):
+        return self.keypoints.shape[0]
+
+    @property
+    def device(self):
+        return self.keypoints.device
+
+    @property
+    def x(self):
+        return self.keypoints[:, 0]
+
+    @property
+    def y(self):
+        return self.keypoints[:, 1]
+
+    def to(self, *args, **kwargs):
+        return DISKFeatures(
+            self.keypoints.to(*args, **kwargs),
+            self.descriptors.to(*args, **kwargs),
+            self.detection_scores.to(*args, **kwargs),
+        )
+
+
+@dataclass
+class Keypoints:
```

Comment:
I don't have a name suggestion, but I think we could avoid that name to avoid confusion with https://github.com/kornia/kornia/blob/master/kornia/geometry/keypoints.py#L13

Reply:
Well, they both are keypoints, and DISK keypoints are not really LAFs (kornia structure for local features), as they lack scale. Renaming them probably would cause even more confusion, so I'd keep them as is.

Reply:
Above, plus this is an internal data structure that the user should never see (it will be merged with descriptors and turned into `DISKFeatures` before the API boundary).